text/0000-core-net-types.md

```diff
@@ -0,0 +1,80 @@
+- Feature Name: `core_net_types`
+- Start Date: 2019-12-06
+- RFC PR: [rust-lang/rfcs#2832](https://github.com/rust-lang/rfcs/pull/2832)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Make the `IpAddr`, `Ipv4Addr`, `Ipv6Addr`, `SocketAddr`, `SocketAddrV4`,
+`SocketAddrV6`, `Ipv6MulticastScope` and `AddrParseError` types available in `no_std`
+contexts by moving them into a `core::net` module.
+
+# Motivation
+[motivation]: #motivation
+
+The motivation here is to provide common types for both `no_std` and `std`
+targets which in turn will ease the creation of libraries based around IP
+addresses. Embedded IoT development is one area where this will be beneficial.
+IP addresses are portable across all platforms and have no external
+dependencies which is in line with the definition of the core library.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `core::net::IpAddr`, `core::net::Ipv4Addr`, `core::net::Ipv6Addr`,
+`core::net::SocketAddr`, `core::net::SocketAddrV4`, `core::net::SocketAddrV6`,
+`core::net::Ipv6MulticastScope` and `core::net::AddrParseError` types are
+available in `no_std` contexts.
+
+Library developers should use `core::net` to implement abstractions in order
+for them to work in `no_std` contexts as well.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Since https://github.com/rust-lang/rust/pull/78802 has been merged, IP and
+socket address types are implemented in ideal Rust layout instead of wrapping
+their corresponding `libc` representation.
+
+Formatting for these types has also been adjusted in
+https://github.com/rust-lang/rust/pull/100625 and
+https://github.com/rust-lang/rust/pull/100640 in order to remove the dependency
+on `std::io::Write`.
+
+This means the types are now platform-agnostic, allowing them to be moved from
+`std::net` into `core::net`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Moving the `std::net` types to `core::net` makes the core library less *minimal*.
```

Comment:
```suggestion
Moving the `std::net` types to `core::net` makes the core library less *seemingly minimal*.

However optimisations relied from the compiler means the dead code types must never be included
in the built binary when not used.
```

I think this could be best clarified here what this means ?

Since the code is optimised away on cases where it is not used as https://github.com/rust-lang/rfcs/pull/2832#issuecomment-1370912300 - This provides testing coverage for https://github.com/rust-lang/rust/pull/104265#issuecomment-1370934670